doofinder.php

```diff
@@ -368,91 +297,79 @@ protected function buildFeedUrl($shop_id, $language, $currency)
             . '&dfsec_hash=' . Configuration::get('DF_API_KEY');
     }
 
-    protected function renderFormCustomCSS($adv = false)
+    /**
+     * Check if the form to create a store installation has to be displayed
+     *
+     * @param Shop $shop
+     *
+     * @return bool
+     */
+    protected function showNewShopForm($shop)
     {
-        $helper = new HelperForm();
-
-        $helper->show_toolbar = false;
-        $helper->table = $this->table;
-        $helper->module = $this;
-        $helper->default_form_language = $this->context->language->id;
-        $helper->allow_employee_form_lang = Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG', 0);
-
-        $helper->identifier = $this->identifier;
-        // $helper->submit_action = 'submitDoofinderModuleCustomCSS';
-        $helper->currentIndex = $this->context->link->getAdminLink('AdminModules', false)
-            . (($adv) ? '&adv=1' : '')
-            . '&configure=' . $this->name . '&tab_module=' . $this->tab . '&module_name=' . $this->name;
-        $helper->token = Tools::getAdminTokenLite('AdminModules');
+        $installation_id = Configuration::get('DF_INSTALLATION_ID', null, (int) $shop->id_shop_group, (int) $shop->id);
+        $multishop_enable = Configuration::get('PS_MULTISHOP_FEATURE_ACTIVE');
+        $apikey = Configuration::get('DF_AI_APIKEY');
 
-        $helper->tpl_vars = [
-            'fields_value' => $this->getConfigFormValuesCustomCSS(),
-            'languages' => $this->context->controller->getLanguages(),
-            'id_language' => $this->context->language->id,
-        ];
-        $this->context->smarty->assign('id_tab', 'custom_css_tab');
-        $html = $this->context->smarty->fetch($this->local_path . 'views/templates/admin/dummy/pre_tab.tpl');
-        $html .= $helper->generateForm([$this->getConfigFormCustomCSS()]);
-        $html .= $this->context->smarty->fetch($this->local_path . 'views/templates/admin/dummy/after_tab.tpl');
+        if (!$installation_id && $multishop_enable && $apikey) {
+            return true;
+        }
```

Comment:
En vez de hacer dos returns, y uno dentro del if, se puede sustituir por esto
```suggestion
        return !$installation_id && $multishop_enable && $apikey
        
        
```

Reply:
Buena :+1:. Modificado